system/prometheus-vmware/templates/_prometheus-vmware-scrape-config.yaml.tpl

```diff
@@ -0,0 +1,368 @@
+# Scrape config for service endpoints.
```

Comment:
Ideally, the below config can be reduced to a minimum using [ServiceMonitors](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#servicemonitor) and [PodMonitors](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#podmonitor).
This allows decoupling the Prometheus server from the scrape configuration: Each scrape target, e.g. exporter can define a ServiceMonitor/PodMonitor with its specific configuration eliminating the need to adjust any central configuration and re-deploy the Prometheus itself.
Most of the targets should be scraped via the generic endpoint, pod SD.
You get them for free as well for any service/pod labeled `prometheus.io/scrape: "true"` and  `prometheus.io/targets: "vmware"` if you do in the values.yaml:
```
prometheus-server:
  serviceDiscoveries:
    endpoints:
      enabled: true
    pods:
      enabled: true
```